home/private_dot_local/lib/agent-equipment/agent_equipment/validator.py

```
@@ -177,11 +301,35 @@ def _load_catalog_lock_with_schemas(
+            catalog_document = _hold_bounded_document(
+                descriptors,
+                Path(catalog_path),
+                maximum_bytes=MAX_CATALOG_BYTES,
+            )
+            lock_document = _hold_bounded_document(
+                descriptors,
+                Path(lock_path),
+                maximum_bytes=MAX_LOCK_BYTES,
+            )
+            if catalog_document.identity[:2] == lock_document.identity[:2]:
+                raise ValueError("catalog and lock must be distinct files")
```

Ruff (0.16.1)

[warning] Abstract `raise` to an inner function

(TRY301)